src/statmodels.jl

```diff
@@ -519,6 +519,56 @@ function show(io::IO, ct::CoefTable)
     nothing
 end
 
+function show(io::IO, ::MIME"text/markdown", ct::CoefTable)
+    cols = ct.cols; rownms = ct.rownms; colnms = ct.colnms;
+    nc = length(cols)
+    nr = length(cols[1])
+    if length(rownms) == 0
+        rownms = [lpad("[$i]",floor(Integer, log10(nr))+3) for i in 1:nr]
+    end
+    mat = [j == 1 ? NoQuote(rownms[i]) :
+           j-1 == ct.pvalcol ? PValue(cols[j-1][i]) :
+           j-1 in ct.teststatcol ? TestStat(cols[j-1][i]) :
+           cols[j-1][i] isa AbstractString ? NoQuote(cols[j-1][i]) : cols[j-1][i]
+           for i in 1:nr, j in 1:nc+1]
+    # Code inspired by print_matrix in Base
+    io = IOContext(io, :compact=>true, :limit=>false)
+    A = Base.alignment(io, mat, 1:size(mat, 1), 1:size(mat, 2),
+                       typemax(Int), typemax(Int), 3)
+    nmswidths = pushfirst!(length.(colnms), 0)
+    A = [nmswidths[i] > sum(A[i]) ? (A[i][1]+nmswidths[i]-sum(A[i]), A[i][2]) : A[i]
+         for i in 1:length(A)]
+    totwidth = sum(sum.(A)) + 2 * (length(A) - 1)
+
+    #   is non breaking space, which means
+    # that we can use the padding logic from the plaintext show method
+    # and get markdown that looks as nice visually as it will render
```

Comment:
I don't understand this part. What have non breaking spaces to do with Markdown?

Reply:
Markdown itself completely ignores non-breaking spaces, but:
- They prevent linebreaks in a plain text editor (but otherwise look like the REPL `show` behavior)
- Depending on how that's converted into HTML, non-breaking spaces *may* be preserved, which would preserve the decimal point alignment (but no guarantees on that front), which would otherwise be impossible in Common Markdown.

I'm happy to change that back to normal spaces. I think it might be more clever than functional. :smile:

Reply:
Mmm, I don't think we want spaces to be preserved when converting to HTML. That would only preserve alignment when using a fixed width font, which isn't very common in HTML, and that would prevent text editors or spreadsheets from treating such columns as numeric.

So yeah I think it would be better not to be too clever here. :-)

Reply:
Cleverness removed. :smile: